app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt

```diff
@@ -336,49 +341,50 @@ open class DefaultToolbarMenu(
     private fun getSyncItemTitle() =
         accountManager.accountProfileEmail ?: context.getString(R.string.sync_menu_sign_in)
 
-    val syncMenuItem = BrowserMenuImageText(
-        getSyncItemTitle(),
-        R.drawable.ic_signed_out,
-        primaryTextColor()
-    ) {
-        onItemTapped.invoke(
-            ToolbarMenu.Item.SyncAccount(accountManager.accountState)
-        )
-    }
-
     @VisibleForTesting(otherwise = PRIVATE)
-    val coreMenuItems by lazy {
-        val menuItems =
-            listOfNotNull(
-                if (shouldUseBottomToolbar) null else menuToolbar,
-                newTabItem,
-                BrowserMenuDivider(),
-                bookmarksItem,
-                historyItem,
-                downloadsItem,
-                extensionsItem,
-                syncMenuItem,
-                BrowserMenuDivider(),
-                getSetDefaultBrowserItem(),
-                getSetDefaultBrowserItem()?.let { BrowserMenuDivider() },
-                findInPageItem,
-                desktopSiteItem,
-                customizeReaderView.apply { visible = ::shouldShowReaderViewCustomization },
-                openInApp.apply { visible = ::shouldShowOpenInApp },
-                reportSiteIssuePlaceholder,
-                BrowserMenuDivider(),
-                addToHomeScreenItem.apply { visible = ::canAddToHomescreen },
-                installToHomescreen.apply { visible = ::canInstall },
-                addToTopSitesItem,
-                saveToCollectionItem,
-                BrowserMenuDivider(),
-                settingsItem,
-                if (shouldDeleteDataOnQuit) deleteDataOnQuit else null,
-                if (shouldUseBottomToolbar) BrowserMenuDivider() else null,
-                if (shouldUseBottomToolbar) menuToolbar else null
+    var coreMenuItems = coreMenuItems()
```

Comment:
You might be able to keep the lazy {} after this refactor

Reply:
Thanks @MozillaNoah
I'll try to keep the `lazy`